compiler/rustc_hir_analysis/src/bounds.rs

```diff
@@ -85,12 +85,44 @@ impl<'tcx> Bounds<'tcx> {
                 }
                 tcx.consts.true_
             }
+            (DefKind::Trait, ty::BoundConstness::ConstIfConst) => {
+                // we are in a trait, where `bound_trait_ref` is a super trait `trait Foo: ~const Bar` or
+                // a where clause `where for<..> Something: ~const Bar`. In these cases, we generate
+                // `<Self as Foo>::Effects: TyCompat<<Self as Bar>::Effects>` and
+                // `for<..> <Self as Foo>::Effects: TyCompat<<Something as Bar>::Effects>`
```

Comment:
Can you like... rearrange these to not read like "if we have A or B, then we do A' or B'" but instead read like "if we have A, we do A'. If we have B, we do B'". I'm quite confused what it's even trying to say.

Reply:
I've reformatted it so that it is clearer now.